### PR TITLE
Make TopicReplicationFactorAnomalyFinder ignore topic with large minISR.

### PR DESCRIPTION
**Pull request by kidkun**
Addresses the issue #1179 .

### CODE REVIEW

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/TopicReplicationFactorAnomalyFinder.java [hunk 1]
```diff
@@ -124,28 +153,33 @@ private void maybeRetrieveAndCacheTopicMinISR(Set<String> topicsToCheck) {
   }
 
   /**
-   * Scan through topics with bad replication factor to check whether the topic is fixable or not.
-   * One topic is fixable if the target replication factor is no less than the topic's minISR config plus topic replication
-   * factor margin.
+   * Scan through topics to check whether the topic having partition(s) with bad replication factor. For each topic, the
+   * target replication factor to check against is the maximal of value of {@link #SELF_HEALING_TARGET_TOPIC_REPLICATION_FACTOR_CONFIG}
```
Comment (efeg) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/TopicReplicationFactorAnomalyFinder.java:157]: `maximal of` -> `maximum`?

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/TopicReplicationFactorAnomalyFinder.java [hunk 2]
```diff
@@ -124,28 +153,33 @@ private void maybeRetrieveAndCacheTopicMinISR(Set<String> topicsToCheck) {
...
         short topicMinISR = _cachedTopicMinISR.get(topic).minISR();
-        if (_targetReplicationFactor < topicMinISR + _topicReplicationFactorMargin) {
-          topicsByFixability.putIfAbsent(false, new HashSet<>());
-          topicsByFixability.get(false).add(topic);
-        } else {
-          topicsByFixability.putIfAbsent(true, new HashSet<>());
-          topicsByFixability.get(true).add(topic);
+        short targetReplicationFactor = (short) Math.max(_targetReplicationFactor, topicMinISR + _topicReplicationFactorMargin);
+        int violatedPartitionCount = 0;
+        for (PartitionInfo pinfo : cluster.partitionsForTopic(topic)) {
```
Comment (efeg) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/TopicReplicationFactorAnomalyFinder.java:170]: Nit: `pinfo` -> `partitionInfo`

File: cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/detector/TopicReplicationFactorAnomalyFinderTest.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+import org.easymock.EasyMock;
+import org.junit.Test;
+
+import static com.linkedin.kafka.cruisecontrol.detector.TopicReplicationFactorAnomalyFinder.DESCRIBE_TOPIC_CONFIG_TIMEOUT_MS;
+import static org.apache.kafka.common.config.TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+
+public class TopicReplicationFactorAnomalyFinderTest {
```
Comment (efeg) [inline cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/detector/TopicReplicationFactorAnomalyFinderTest.java:36]: Missing `EasyMock.verify()` in tests.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/TopicReplicationFactorAnomalyFinder.java [hunk 4]
```diff
@@ -59,12 +64,12 @@
   public static final String DEFAULT_TOPIC_EXCLUDED_FROM_REPLICATION_FACTOR_CHECK = "";
   public static final String TOPIC_REPLICATION_FACTOR_ANOMALY_CLASS_CONFIG = "topic.replication.topic.anomaly.class";
   public static final Class<?> DEFAULT_TOPIC_REPLICATION_FACTOR_ANOMALY_CLASS = TopicReplicationFactorAnomaly.class;
-  public static final String TOPICS_WITH_BAD_REPLICATION_FACTOR_BY_FIXABILITY_CONFIG = "topics.with.bad.replication.factor.by.fixability";
+  public static final String BAD_TOPICS_BY_REPLICATION_FACTOR_CONFIG = "bad.topics.by.replication.factor";
```
[the remark this answers is not in the document]
Comment (kidkun) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/TopicReplicationFactorAnomalyFinder.java:67]: This is not a config of this class, it is a config name used to pass value to `TopicReplicationFactorAnomaly`.

**Pull request merged**